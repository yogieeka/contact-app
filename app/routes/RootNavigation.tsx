/**
 * This is a Navigation file which is wired already with Bottom Tab Navigation.
 * If you don't like it, feel free to replace with your own setup.
 * Uncomment commented lines from return() of RootNavigation to wire Login flow
 */
import React from 'react';
import {ColorValue} from 'react-native';

import {NavigationContainer} from '@react-navigation/native';
import {createBottomTabNavigator} from '@react-navigation/bottom-tabs';
import Icon from 'react-native-vector-icons/Ionicons';

import {useTheme} from '../theme/useTheme';
import {typeVariants} from '../theme/theme';

// Screens
import Tasks from '../screens/Tasks';
import NetworkExample from '../screens/NetworkExample';
import Settings from '../screens/Settings';

// Icons for Bottom Tab Navigation
const homeIcon = ({color}: {color: ColorValue | number}) => (
  <Icon name="list-sharp" size={30} color={color} />
);
const networkIcon = ({color}: {color: ColorValue | number}) => (
  <Icon name="wifi-sharp" size={24} color={color} />
);
const settingsIcon = ({color}: {color: ColorValue | number}) => (
  <Icon name="settings-sharp" size={24} color={color} />
);

// Root Navigation
// const Stack = createNativeStackNavigator();
const Tab = createBottomTabNavigator();

export default function RootNavigation() {
  const {theme} = useTheme();

  return (
    <NavigationContainer>
      <Tab.Navigator
        screenOptions={{
          tabBarStyle: {
            backgroundColor: theme.cardBg,
            borderTopColor: theme?.layoutBg,
          },
          tabBarInactiveTintColor: theme.color,
          tabBarActiveTintColor: theme.primary,
          headerStyle: {backgroundColor: theme.cardBg, height: 50},
          headerTitleAlign: 'center',
          headerTitleStyle: {
            fontFamily: typeVariants.titleLarge.fontFamily,
            fontSize: 18,
            color: theme.primary,
            fontWeight: 'bold',
          },
          tabBarShowLabel: false,
        }}>
        <Tab.Screen
          name="Contact"
          component={Tasks}
          options={{
            tabBarIcon: homeIcon,
            tabBarTestID: 'BottomTab.ToDo',
          }}
        />
        <Tab.Screen
          name="NetworkExample"
          component={NetworkExample}
          options={{
            tabBarIcon: networkIcon,
            tabBarTestID: 'BottomTab.Network',
          }}
        />
        <Tab.Screen
          name="Settings"
          component={Settings}
          options={{
            // headerShown: false,
            tabBarIcon: settingsIcon,
            tabBarTestID: 'BottomTab.Settings',
          }}
        />
      </Tab.Navigator>
    </NavigationContainer>
  );
}
