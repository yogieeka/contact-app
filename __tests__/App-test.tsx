/**
 * @format
 * @lint-ignore-every XPLATJSCOPYRIGHT1
 */
import 'react-native';
import React from 'react';
// Note: import explicitly to use the types shipped with jest.
import {jest, describe, it, expect} from '@jest/globals';
// import {render} from '@testing-library/react-native';

import {render} from '../app/utils/test-utils';

import App from '../App';
import RootNavigation from '../app/routes/RootNavigation';

jest.useFakeTimers();

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

describe('App', () => {
  it('renders correctly', () => {
    renderer.create(<App />);
  });

  it('should match snapshot correctly', () => {
    const page = render(<RootNavigation />, {});
    expect(page.toJSON()).toMatchSnapshot();
  });

  it('Renders bottom tab bar', () => {
    let {getByTestId} = render(<RootNavigation />, {});
    getByTestId('BottomTab.Network');
  });
});
