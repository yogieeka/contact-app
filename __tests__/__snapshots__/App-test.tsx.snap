// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenNavigationContainer
    style={
      {
        "flex": 1,
        "overflow": "hidden",
      }
    }
  >
    <RNSScreen
      activityState={2}
      collapsable={false}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      onGestureCancel={[Function]}
      sheetAllowedDetents="large"
      sheetCornerRadius={-1}
      sheetExpandsWhenScrolledToEdge={true}
      sheetGrabberVisible={false}
      sheetLargestUndimmedDetent="all"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "zIndex": 0,
        }
      }
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          [
            {
              "backgroundColor": "rgb(242, 242, 242)",
              "flex": 1,
            },
            [
              {
                "flex": 1,
                "flexDirection": "column-reverse",
              },
              undefined,
            ],
          ]
        }
      >
        <View
          style={
            {
              "flex": 1,
            }
          }
        >
          <RNCSafeAreaView
            edges={
              {
                "bottom": "additive",
                "left": "additive",
                "right": "additive",
                "top": "additive",
              }
            }
            style={
              {
                "flex": 1,
              }
            }
            testID="Screen.Tasks"
          >
            <View
              style={
                [
                  {
                    "backgroundColor": "#ffffff",
                    "flex": 1,
                  },
                  {
                    "backgroundColor": "#e0eeec",
                  },
                  undefined,
                ]
              }
              testID="Layout.LayoutContainer"
            >
              <RCTScrollView
                contentContainerStyle={
                  {
                    "paddingHorizontal": 12,
                    "paddingVertical": 30,
                  }
                }
                data={[]}
                getItem={[Function]}
                getItemCount={[Function]}
                keyExtractor={[Function]}
                onContentSizeChange={[Function]}
                onLayout={[Function]}
                onMomentumScrollBegin={[Function]}
                onMomentumScrollEnd={[Function]}
                onScroll={[Function]}
                onScrollBeginDrag={[Function]}
                onScrollEndDrag={[Function]}
                refreshControl={
                  <RefreshControlMock
                    colors={
                      [
                        "#2bbca2",
                      ]
                    }
                    onRefresh={[Function]}
                    refreshing={false}
                  />
                }
                removeClippedSubviews={false}
                renderItem={[Function]}
                scrollEventThrottle={0.0001}
                stickyHeaderIndices={[]}
                viewabilityConfigCallbackPairs={[]}
              >
                <RCTRefreshControl />
                <View />
              </RCTScrollView>
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "#2bbca2",
                    "borderRadius": 100,
                    "bottom": 14,
                    "elevation": 10,
                    "height": 60,
                    "justifyContent": "center",
                    "opacity": 1,
                    "position": "absolute",
                    "right": 14,
                    "width": 60,
                  }
                }
              >
                <Text
                  allowFontScaling={false}
                  selectable={false}
                  style={
                    [
                      {
                        "color": "white",
                        "fontSize": 30,
                      },
                      undefined,
                      {
                        "fontFamily": "Ionicons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      {},
                    ]
                  }
                >
                  
                </Text>
              </View>
            </View>
          </RNCSafeAreaView>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            collapsable={false}
            pointerEvents="box-none"
            style={
              {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "zIndex": 0,
              }
            }
          >
            <View
              collapsable={false}
              style={
                {
                  "backgroundColor": "#ffffff",
                  "borderBottomColor": "rgb(216, 216, 216)",
                  "flex": 1,
                  "shadowColor": "rgb(216, 216, 216)",
                  "shadowOffset": {
                    "height": 0.5,
                    "width": 0,
                  },
                  "shadowOpacity": 0.85,
                  "shadowRadius": 0,
                }
              }
            />
          </View>
          <View
            collapsable={false}
            pointerEvents="box-none"
            style={
              {
                "height": 50,
                "maxHeight": undefined,
                "minHeight": undefined,
                "opacity": undefined,
                "transform": undefined,
              }
            }
          >
            <View
              pointerEvents="none"
              style={
                {
                  "height": 0,
                }
              }
            />
            <View
              pointerEvents="box-none"
              style={
                {
                  "alignItems": "stretch",
                  "flex": 1,
                  "flexDirection": "row",
                }
              }
            >
              <View
                collapsable={false}
                pointerEvents="box-none"
                style={
                  {
                    "alignItems": "flex-start",
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "justifyContent": "center",
                    "marginStart": 0,
                  }
                }
              />
              <View
                collapsable={false}
                pointerEvents="box-none"
                style={
                  {
                    "justifyContent": "center",
                    "marginHorizontal": 16,
                    "maxWidth": 718,
                  }
                }
              >
                <Text
                  accessibilityRole="header"
                  aria-level="1"
                  collapsable={false}
                  numberOfLines={1}
                  style={
                    {
                      "color": "#2bbca2",
                      "fontFamily": "Poppins-Bold",
                      "fontSize": 18,
                      "fontWeight": "bold",
                    }
                  }
                >
                  Contact
                </Text>
              </View>
              <View
                collapsable={false}
                pointerEvents="box-none"
                style={
                  {
                    "alignItems": "flex-end",
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "justifyContent": "center",
                    "marginEnd": 0,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
    </RNSScreen>
  </RNSScreenNavigationContainer>
  <View
    collapsable={false}
    onLayout={[Function]}
    pointerEvents="auto"
    style={
      {
        "backgroundColor": "#ffffff",
        "borderTopColor": "#e0eeec",
        "borderTopWidth": 0.5,
        "bottom": 0,
        "elevation": 8,
        "height": 49,
        "left": 0,
        "paddingBottom": 0,
        "paddingHorizontal": 0,
        "position": null,
        "right": 0,
        "transform": [
          {
            "translateY": 0,
          },
        ],
      }
    }
  >
    <View
      pointerEvents="none"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
    <View
      accessibilityRole="tablist"
      style={
        {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        accessibilityLabel="Contact, tab, 1 of 3"
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": true,
          }
        }
        accessibilityStates={
          [
            "selected",
          ]
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "flex": 1,
            },
            {
              "backgroundColor": "transparent",
            },
            {
              "flexDirection": "column",
              "justifyContent": "flex-end",
            },
            undefined,
          ]
        }
        testID="BottomTab.ToDo"
      >
        <View
          style={
            [
              {
                "flex": 1,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "minWidth": 25,
                  "position": "absolute",
                  "width": "100%",
                },
                {
                  "opacity": 1,
                },
              ]
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                [
                  {
                    "color": "#2bbca2",
                    "fontSize": 30,
                  },
                  undefined,
                  {
                    "fontFamily": "Ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  {},
                ]
              }
            >
              
            </Text>
          </View>
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "minWidth": 25,
                  "position": "absolute",
                  "width": "100%",
                },
                {
                  "opacity": 0,
                },
              ]
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                [
                  {
                    "color": "#695D5D",
                    "fontSize": 30,
                  },
                  undefined,
                  {
                    "fontFamily": "Ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  {},
                ]
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
      <View
        accessibilityLabel="NetworkExample, tab, 2 of 3"
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": false,
          }
        }
        accessibilityStates={[]}
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "flex": 1,
            },
            {
              "backgroundColor": "transparent",
            },
            {
              "flexDirection": "column",
              "justifyContent": "flex-end",
            },
            undefined,
          ]
        }
        testID="BottomTab.Network"
      >
        <View
          style={
            [
              {
                "flex": 1,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "minWidth": 25,
                  "position": "absolute",
                  "width": "100%",
                },
                {
                  "opacity": 0,
                },
              ]
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                [
                  {
                    "color": "#2bbca2",
                    "fontSize": 24,
                  },
                  undefined,
                  {
                    "fontFamily": "Ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  {},
                ]
              }
            >
              
            </Text>
          </View>
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "minWidth": 25,
                  "position": "absolute",
                  "width": "100%",
                },
                {
                  "opacity": 1,
                },
              ]
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                [
                  {
                    "color": "#695D5D",
                    "fontSize": 24,
                  },
                  undefined,
                  {
                    "fontFamily": "Ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  {},
                ]
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
      <View
        accessibilityLabel="Settings, tab, 3 of 3"
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": false,
          }
        }
        accessibilityStates={[]}
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "flex": 1,
            },
            {
              "backgroundColor": "transparent",
            },
            {
              "flexDirection": "column",
              "justifyContent": "flex-end",
            },
            undefined,
          ]
        }
        testID="BottomTab.Settings"
      >
        <View
          style={
            [
              {
                "flex": 1,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "minWidth": 25,
                  "position": "absolute",
                  "width": "100%",
                },
                {
                  "opacity": 0,
                },
              ]
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                [
                  {
                    "color": "#2bbca2",
                    "fontSize": 24,
                  },
                  undefined,
                  {
                    "fontFamily": "Ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  {},
                ]
              }
            >
              
            </Text>
          </View>
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "minWidth": 25,
                  "position": "absolute",
                  "width": "100%",
                },
                {
                  "opacity": 1,
                },
              ]
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                [
                  {
                    "color": "#695D5D",
                    "fontSize": 24,
                  },
                  undefined,
                  {
                    "fontFamily": "Ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  {},
                ]
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
